lldb/source/Plugins/LanguageRuntime/Swift/SwiftLanguageRuntimeDynamicTypeResolution.cpp

```diff
@@ -2850,8 +2850,10 @@ SwiftLanguageRuntimeImpl::GetTypeRef(CompilerType type,
     return nullptr;
   swift::Demangle::NodePointer node =
       TypeSystemSwiftTypeRef::GetCanonicalDemangleTree(
-          module_holder ? module_holder : ts->GetSwiftASTContext(), dem,
-          mangled_name.GetStringRef());
+          module_holder ? module_holder : &ts->GetTypeSystemSwiftTypeRef(),
+          module_holder ? module_holder->GetSwiftASTContext()
+                        : ts->GetSwiftASTContext(),
+          dem, mangled_name.GetStringRef());
```

Comment:
I remember these ternaries, I wanted to get back to them to remove them. There are two callers of `SwiftLanguageRuntimeImpl::GetTypeRef()`, and in both cases, they pass in a `type`, and a `TypeSystemSwiftTypeRef` derived from that type. Then here, it's derived again.

In other words, the both caller and callee are doing `type.GetTypeSystem()->GetTypeSystemSwiftTypeRef()` (ignoring the cast for simplicity). I think these ternaries can be removed.